pages/01_Datahub.py

```diff
@@ -0,0 +1,151 @@
+import streamlit as st 
+import pyodbc 
+import pandas as pd
+import time
+from Utils.more_info import markdown_table1
+from Utils.more_info import markdown_table2
+
+st.set_page_config(
+    page_title= "Data Hub", layout="wide"
+)
+
+st.title('Customer Churn Analysis')
+
+
+@st.cache_resource(show_spinner='Please wait for a second...')
+
+def init_connection():
+    return pyodbc.connect(
+        "DRIVER={SQL Server};SERVER="
+        + st.secrets["DBS"]
+        + ";DATABASE="
+        + st.secrets["DBN"]
+        + ";UID="
+        + st.secrets["DBU"]
+        + ";PWD="
+        + st.secrets["DBP"]
+    )
+
+conn = init_connection()
+
+
+@st.cache_data(show_spinner='Loading data, This will only take a minute..')
+def running_query(query):
+    with conn.cursor() as c:
+        c.execute(query)
+        rows = c.fetchall()
+        df=pd.DataFrame.from_records(rows, columns=[column[0] for column in c.description])
+    return df
+
+def get_all_columns():
+    sql_query = 'SELECT * FROM dbo.LP2_Telco_churn_first_3000'
+    df = running_query(sql_query)
+    return df
+
+
+def load_and_concat_datasets(first_dataset_func, second_dataset_path):
+    """
+    Load the first dataset using a provided function and the second dataset from a CSV file,
+    then concatenate them into a single DataFrame.
+    
+    Parameters:
+    - first_dataset_func: function that returns a DataFrame for the first dataset.
+    - second_dataset_path: string path to the second dataset CSV file.
+    
+    Returns:
+    - A concatenated DataFrame containing both datasets.
+    """
+    # Load the first dataset using the provided function
+    first_train = first_dataset_func()
+    
+    # Load the second dataset from the provided CSV file path
+    second_train = pd.read_csv(second_dataset_path)
+    
+    # Concatenate the two DataFrames
+    train_df = pd.concat([first_train, second_train], ignore_index=True)
+    
+    return train_df
+
+train_df = load_and_concat_datasets(get_all_columns, "./Datasets/LP2_Telco-churn-second-2000.csv")
+
+
```

Comment:
The code is well written. But I think you should limit the mapping values to just Yes or No rather than Yes, No, and No phone service/No internet service.